provider/infoblox.go

```diff
@@ -0,0 +1,384 @@
+	}
+
+	httpPoolConnections, _ := strconv.ParseInt(getEnv("HTTP_POOL_CONNECTIONS", "10"), 0, 0)
+	httpRequestTimeout, _ := strconv.ParseInt(getEnv("HTTP_REQUEST_TIMEOUT", "60"), 0, 0)
+
+	transportConfig := ibclient.NewTransportConfig(
+		getEnv("SSL_VERIFY", "true"),
+		int(httpRequestTimeout),
+		int(httpPoolConnections),
+	)
+
+	requestBuilder := &ibclient.WapiRequestBuilder{}
+	requestor := &ibclient.WapiHttpRequestor{}
+
+	client, err := ibclient.NewConnector(hostConfig, transportConfig, requestBuilder, requestor)
+
+	if err != nil {
+		logrus.Fatal(err)
+		return nil, err
+	}
+
+	provider := &InfobloxProvider{
+		client:       client,
+		domainFilter: domainFilter,
+		dryRun:       dryRun,
+	}
+
+	return provider, nil
+}
+
+// Records gets the current records.
+func (p *InfobloxProvider) Records() (endpoints []*endpoint.Endpoint, err error) {
+	zones, err := p.zones()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, zone := range zones {
+		var resA []ibclient.RecordA
+		objA := ibclient.NewRecordA(
+			ibclient.RecordA{
+				Zone: zone.Fqdn,
+			},
+		)
+		err = p.client.GetObject(objA, "", &resA)
+		if err != nil {
+			return nil, err
+		}
+		for _, res := range resA {
+			endpoints = append(endpoints, endpoint.NewEndpoint(res.Name, res.Ipv4Addr, endpoint.RecordTypeA))
+		}
+
+		var resC []ibclient.RecordCNAME
+		objC := ibclient.NewRecordCNAME(
+			ibclient.RecordCNAME{
+				Zone: zone.Fqdn,
+			},
+		)
+		err = p.client.GetObject(objC, "", &resC)
+		if err != nil {
+			return nil, err
+		}
+		for _, res := range resC {
+			endpoints = append(endpoints, endpoint.NewEndpoint(res.Name, res.Canonical, endpoint.RecordTypeCNAME))
+		}
+
+		var resT []ibclient.RecordTXT
+		objT := ibclient.NewRecordTXT(
+			ibclient.RecordTXT{
+				Zone: zone.Fqdn,
+			},
+		)
+		err = p.client.GetObject(objT, "", &resT)
+		if err != nil {
+			return nil, err
+		}
+		for _, res := range resT {
+			if _, err := strconv.Unquote(res.Text); err != nil {
+				res.Text = strconv.Quote(res.Text)
+			}
+			endpoints = append(endpoints, endpoint.NewEndpoint(res.Name, res.Text, endpoint.RecordTypeTXT))
+		}
+	}
+	return endpoints, nil
+}
+
+// ApplyChanges applies the given changes.
+func (p *InfobloxProvider) ApplyChanges(changes *plan.Changes) error {
+	zones, err := p.zones()
+	if err != nil {
+		return err
+	}
+
+	created, deleted, updated := p.mapChanges(zones, changes)
+	p.createRecords(created)
+	p.deleteRecords(deleted)
+	p.updateRecords(updated)
+	return nil
+}
+
+func (p *InfobloxProvider) zones() ([]ibclient.ZoneAuth, error) {
+	var res, result []ibclient.ZoneAuth
+	obj := ibclient.NewZoneAuth(ibclient.ZoneAuth{})
+	err := p.client.GetObject(obj, "", &res)
+
+	if err != nil || res == nil || len(res) == 0 {
+		return nil, err
+	}
+
+	for _, zone := range res {
+		if p.domainFilter.Match(zone.Fqdn) {
+			result = append(result, zone)
+		}
+	}
+
+	return result, nil
+}
+
+type infobloxChangeMap map[*ibclient.ZoneAuth][]*endpoint.Endpoint
```

Comment:
I'll switch the key to `string`, with `zone.Fqdn` as the value. I borrowed this idea from the [Azure provider](https://github.com/kubernetes-incubator/external-dns/blob/f14a953b5faa2b22030e9db28954f5fa0656497f/provider/azure.go#L235), which it would seem is susceptible to the same concern.

Reply:
:+1: 

Actually, the way the map is used it's not an issue, but when you can just use `zone.Fqdn` here then  it's less confusing code IMO.